browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -62,19 +110,93 @@ bool IPFSTabHelper::MaybeCreateForWebContents(
   return true;
 }
 
+void IPFSTabHelper::HostResolvedCallback(
+    const std::string& host,
+    const std::vector<std::string>& text_results) {
+  std::string dnslink = GetDSNRecordValue(text_results, kDnsLinkHeader);
+  if (dnslink.empty())
```

Comment:
i think we want to add more validation here before passing the TXT value to IPFS. there are certain URLs that websites are not allowed to open for security reasons, for instance `file:///etc/hosts` or `chrome://settings`. if an attacker creates a site with a dnslink value set to a privileged URL like this, it should not open.